tests/testthat/test-tamper.R

```diff
@@ -0,0 +1,82 @@
+
+context("Tampering")
+
+test_that("We can detect if we are in a pipe", {
+
+  skip(message = "not ready yet for a real test")
```

Comment:
I think if we write `dump_pipes()` and test that, we'll be fine. No real need to test the `menu` function, right? We can of course still test what `menu` would print out. We just need to break this up in small functions, exactly as you said before, and it is actually very possible, and then test the non-interactive pieces.